### ISSUE TITLE
Implement proper grabbing behavior on macOS

### ISSUE THREAD

**Issue by Osspial** [rust-windowing/winit#1093]
Right now on macOS, the cursor just gets locked to its current location, instead of getting locked to the area of the window. A proper solution that locks the cursor to the window exists and is documented, but hasn't been implemented yet.

See this comment:

https://github.com/rust-windowing/winit/blob/2442305bb7f5e0e9c95733536e5153129dde59db/src/platform_impl/macos/window.rs#L519-L523

**Comment by panda2134** [comment]
Can anyone merge this PR? This feature is critical for developing cross-platform video games. If we cannot set mouse grab on macOS, then camera movement will become quite weird there.

**Comment by kchibisov** [comment]
@panda2134 this is not a PR, it's an issue. macOS has locking, but not confining.

**Comment by panda2134** [comment]
Sorry for mistaking this as a PR. If it's the OS's fault, then that's acceptable limitation. Also I just found out that it works when locking is used after setting the cursor to somewhere in the window.

**Comment by kchibisov** [comment]
Yes, the cursor should be in the window to lock it. that's how it works on most OSes.

**Comment by swiftcoder** [comment]
You can simulate proper cursor grabbing with the cursor lock API, by only locking after the user first clicks in the window.

(I believe this is required for Web anyway, because you can only grab the cursor in response to user interaction?)
